single_tech_samples/datafactory/scripts/deploy_infrastructure.sh

```diff
@@ -0,0 +1,213 @@
+# - Correct Azure subscription is selected
+#######################################################
+
+set -o errexit
+set -o pipefail
+set -o nounset
+# set -o xtrace # For debugging
+
+###################
+# REQUIRED ENV VARIABLES:
+#
+# ENV_NAME
+# RESOURCE_GROUP_NAME
+# RESOURCE_GROUP_LOCATION
+# AZURE_SUBSCRIPTION_ID
+
+
+#####################
+# DEPLOY ARM TEMPLATE
+
+# Set account to where ARM template will be deployed to
+echo "Deploying to Subscription: $AZURE_SUBSCRIPTION_ID"
+az account set --subscription $AZURE_SUBSCRIPTION_ID
+
+# Create resource group
+echo "Creating resource group: $RESOURCE_GROUP_NAME"
+az group create --name "$RESOURCE_GROUP_NAME" --location "$RESOURCE_GROUP_LOCATION" --tags Environment=$ENV_NAME
+
+# By default, set all KeyVault permission to deployer
+# Retrieve KeyVault User Id
+kv_owner_object_id=$(az ad signed-in-user show --output json | jq -r '.objectId')
+
+# Deploy arm template
+echo "Deploying resources into $RESOURCE_GROUP_NAME"
+arm_output=$(az deployment group create \
+    --resource-group "$RESOURCE_GROUP_NAME" \
+    --template-file "./infrastructure/azuredeploy.json" \
+    --parameters @"./infrastructure/azuredeploy.parameters.${ENV_NAME}.json" \
+    --parameters keyvault_owner_object_id=${kv_owner_object_id} deployment_id=${DEPLOYMENT_ID} \
+    --output json)
+
+if [[ -z $arm_output ]]; then
+    echo >&2 "ARM deployment failed."
+    exit 1
+fi
+
+#########################
+# CONFIGURE DATA LAKE
+
+# Retrieve storage account name
+dl_storage_account=$(echo $arm_output | jq -r '.properties.outputs.datalake_storage_account_name.value')
+export DL_STORAGE_ACCOUNT=$dl_storage_account
+
+# Retrieve storage account (ADLS Gen2) key
+dl_storage_key=$(az storage account keys list \
+    --account-name $dl_storage_account \
+    --resource-group $RESOURCE_GROUP_NAME \
+    --output json |
+    jq -r '.[0].value')
+export DL_STORAGE_KEY=$dl_storage_key
+
+# Retrieve full storage account azure id
+dl_stor_id=$(az storage account show \
+    --name "$dl_storage_account" \
+    --resource-group "$RESOURCE_GROUP_NAME" \
+    --output json |
+    jq -r '.id')
+
+# Add file system storage account
+storage_file_system=datalake
+echo "Creating ADLS Gen2 File system: $storage_file_system"
+az storage container create --account-name $dl_storage_account --account-key $dl_storage_key --name $storage_file_system
+
+echo "Creating folders within the file system."
+declare -a zones=("/dropzone" "/bronze" "/silver" "/gold")
+for zone in "${zones[@]}"
+do
+    az storage fs directory create --account-name $dl_storage_account --account-key $dl_storage_key -n $zone -f $storage_file_system
+done
+
+# Get Azure Data Factory managed service identity
+export DATAFACTORY_NAME=$(echo $arm_output | jq -r '.properties.outputs.datafactory_name.value')
+adf_msi=$(az resource show \
+         --name $DATAFACTORY_NAME \
+         --resource-group $RESOURCE_GROUP_NAME \
+         --resource-type "Microsoft.DataFactory/factories" |
```

Comment:
Add `--output json` to az command. Without this, it will fail for users who have configured their default az output to be something else (ei. table) other than json.

Reply:
Thank you, fixed this